### PR TITLE
SVCS-802 Accept oAuth1/2 tokens in config

### PR DESCRIPTION
**Pull request by wwaaijer-exh**
Might be easier to review the commits separately.
While making these changes it became clear some tests were either incomplete or misleading and needed some refactoring.

### CODE REVIEW

File: src/http/oauth2.ts [hunk 1]
```diff
@@ -103,7 +110,7 @@ export function createOAuth2HttpClient(
     ) {
       originalRequest.isRetryWithRefreshedTokens = true;
       await refreshTokens();
-      return http(originalRequest);
+      return httpWithAuth(originalRequest);
```
Comment (wwaaijer-exh) [inline src/http/oauth2.ts:113]: This actually solved a bug introduced in the current version we're working on. Was caught after correcting the test suite, so was definitely worth it already 😬

File: tests/unit/client/oauth2.test.ts [hunk 2]
```diff
@@ -244,14 +288,23 @@ describe('OAuth2HttpClient', () => {
...
-      await confidentialHttpWithAuth.extraAuthMethods.authenticate(mockAuth);
+      await confidentialHttpWithAuth.extraAuthMethods.authenticate(
+        emailAuthData
+      );
+    expect(authenticateResult).toStrictEqual({ accessToken: mockToken });
+
+    // Expect successive requests to be authenticated
     nock(mockParams.host).get('/test').reply(200, '');
 
     const result = await confidentialHttpWithAuth.get('test');
```
Comment (chad-pelly) [inline tests/unit/client/oauth2.test.ts:305]: (minor) Could we randomHexString / uuid or something: `mockToken`, we have `test`, `/test` and `test` and it makes the context a little harder to read 🤔 

Same for clientId and secret, could we make them the same references if they're related objects?

File: tests/unit/client/oauth2.test.ts [hunk 3]
```diff
@@ -204,34 +245,37 @@ describe('OAuth2HttpClient', () => {
         },
       });
 
+    let mfaError;
+    try {
```
Comment (chad-pelly) [inline tests/unit/client/oauth2.test.ts:249]: (opinion / ignore) Just for cleanliness  as this is a large test, I would 
`await httpWithAuth.extraAuthMethods.authenticate(emailAuthData).catch(error => {
      mfaError = error;
 });`

**Review by chad-pelly — APPROVED** [review]
Looks great, a few opinions / minors, but nothing blocking at all 👌

File: tests/unit/client/oauth2.test.ts (hunk 2 above)
Comment (wwaaijer-exh) [reply tests/unit/client/oauth2.test.ts:305]: 💯 Will update

**Review by chad-pelly — APPROVED** [review]
:white_check_mark: :saluting_face: :muscle::skin-tone-3: :+1::skin-tone-3: :heavy_check_mark: :ballot_box_with_check: :rocket:

**Pull request merged**